Add in CONTRIBUTING.md to clarify open source modification

chriswebb09 opened the pull request:
Update to clarify issues regarding modifications to the toolkit.

chriswebb09 committed:
update readme with clarification

chriswebb09 committed:
update wording

dfeinzimer commented:
We might look to see if other toolkits have similar messaging and match them on this. My feeling is that the concept of forking should be somewhat presumed knowledge for GitHub users.

yo1995 reviewed (commented):
What David asked is reasonable. We should check other SDK's toolkit's verbiage

- https://github.com/Esri/arcgis-maps-sdk-kotlin-toolkit?tab=readme-ov-file#contributing
- https://github.com/Esri/arcgis-maps-sdk-dotnet-toolkit?tab=readme-ov-file#contribute

I'm still in favor of adding some words that cover the points in #1287 . At minimum, I think what dotnet has is a good addition.

Ideally, it should tell the following things

- one can directly copy and paste source code from the toolkit instead of importing the whole package
- one can fork the repo and make their own edits and use it as a Swift Package
    - with the possibility of keeping up with the upstream repo
- contributions are welcome

chriswebb09 commented on README.md:
```diff
@@ -89,6 +88,8 @@ Please see [Configure App Secrets](Documentation/ConfigureAppSecrets.md) for add
 
 Find a bug or want to request a new feature?  Please let us know by [submitting an issue](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/issues/new).
 
+Because this project is meant to be used for a large variety of users and use cases, we are unable to accomodate every request for changes that is made. However, fear not, as an open source project you should feel free to create your own fork of the repository and modify it for your specific use case. 
```
@mhdostal @yo1995 so remove this and add in Contributing.md page?

dfeinzimer reviewed (commented):
`Contributing.md` sounds like the right direction.

chriswebb09 committed:
update README.md

dfeinzimer reviewed (commented):
Consider updating the title of the PR for future reference.

chriswebb09 committed:
Apply suggestions from code review

Co-authored-by: David Feinzimer <dfeinzimer@esri.com>

yo1995 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,48 @@
+### Submitting changes
+
+1. Push the changes in your feature branch to your repo.
+2. Submit a [pull request](https://help.github.com/articles/using-pull-requests) against the "main" branch.
```
`v.next` or `main` branch? I think this is one of the reasons we don't encourage users to make PRs to our repos - sometimes `v.next` contains code that depend on new APIs. I'm OK if we can correctly manage the user submitted changes on `main` - after all I've never seen a user submitted bug fix so far…

chriswebb09 committed:
Apply suggestions from code review

Co-authored-by: Ting <tchen@esri.com>

chriswebb09 committed:
fix License

chriswebb09 committed:
update with v.next

chriswebb09 committed:
update README

chriswebb09 commented on README.md:
```diff
@@ -34,6 +34,10 @@ To use Toolkit in your project:
 
+### Standalone Components
+
+If you want to use a specific component in your project, you can copy and paste the source code directly into your project without using a the Swift Package Manager.
```
@yo1995 Something like this for the copy and paste?

chriswebb09 commented:
@yo1995 
> * contributions are welcome

In the README there is a section on contributing, do we need to do anything additional? 

> ## Contributing
> 
> Esri welcomes contributions from anyone and everyone. Please see our [guidelines for contributing](https://github.com/esri/contributing).

mhdostal commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,48 @@
+# Contribution Guidelines
```
This file isn't referenced by `README.md`. Is it necessary, or is the one referenced below ([guidelines for contributing](https://github.com/esri/contributing)) sufficient?

The title for this section is "Contribution Guidelines", but then the section paragraph states _see our guidelines for contributing..._, which seems redundant.

I would be fine just referencing the main esri guidelines (https://github.com/esri/contributing).

chriswebb09 committed:
Revise contribution guidelines for sprint review discussion

Updated the contribution guidelines to welcome contributions from anyone and everyone.

chriswebb09 committed:
Update README.md

Co-authored-by: Ting <tchen@esri.com>

chriswebb09 committed:
Update copyright year to 2026

dfeinzimer commented on README.md:
```diff
@@ -35,6 +34,10 @@ To use Toolkit in your project:
 
+### Standalone Components
+
+If you want to use a specific component in your project, you can copy and paste the source code directly into your project without using the Swift Package Manager.
```
```suggestion
If you want to use a specific component in your project, you can copy and paste the source code directly into your project without using the Swift Package Manager version.
```
or

```suggestion
If you want to use a specific component in your project, you can copy and paste the source code directly into your project without using the official package.
```
or just

```suggestion
If you want to use a specific component in your project, you can copy and paste the source code directly into your project without using the package.
```

chriswebb09 committed:
Apply suggestions from code review

Co-authored-by: David Feinzimer <dfeinzimer@esri.com>

dfeinzimer commented on README.md:
```diff
@@ -35,6 +34,10 @@ To use Toolkit in your project:
 
 ## Instructions
 
+### Standalone Components
+
+If you want to use a specific component in your project, you can copy and paste the source code directly into your project without using the Swift Package Manager.
+
```
Something like the following would seemingly be sufficient:

> If you want to use a specific component in your project, you may copy and paste the source code. Retain all copyright notices and include a copy of the Apache 2.0 License.

dfeinzimer commented:
@chriswebb09 let me know if you need any assistance moving this forward, happy to help!

mhdostal commented:
This looks pretty close to being done. I think it's just @dfeinzimer 's last suggestion [here](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/pull/1289#pullrequestreview-3668080405). Let's get this done for the v300 release!

chriswebb09 committed:
Update instructions for using standalone components

Clarified instructions for using specific components by emphasizing the need to retain copyright notices and include the Apache 2.0 License.

chriswebb09 commented:
@mhdostal
